tests/suites/helpers.function

```diff
@@ -69,15 +74,166 @@ typedef struct data_tag
+ *
+ * \param   PARAM_ERROR_VALUE   The expected error code.
+ *
+ * \param   TEST                The test expression to be tested.
+ */
+#define TEST_INVALID_PARAM_RET( PARAM_ERR_VALUE, TEST ) \
+    do {                                                \
+        if ( setjmp( param_fail_jmp ) == 0 )            \
+        {                                               \
+            if( (TEST) != PARAM_ERR_VALUE)              \
+            {                                           \
+                test_fail( #TEST, __LINE__, __FILE__ ); \
+                goto exit;                              \
+            }                                           \
+       }                                                \
+       memset( param_fail_jmp, 0, sizeof(jmp_buf) );    \
     } while( 0 )
 
+/**
+ * \brief   This macro tests the statement passed to it as a test step or
+ *          individual test in a test case. The macro assumes the test will fail
+ *          and will generate an error.
+ *
+ *          It assumes the library function under test cannot return a value and
+ *          assumes errors can only be indicated byt calls to
+ *          MBEDTLS_PARAM_FAIL.
+ *
+ *          When MBEDTLS_CHECK_PARAMS is enabled, calls to the parameter failure
+ *          callback, MBEDTLS_PARAM_FAIL, are assumed to indicate the
+ *          expected failure. If MBEDTLS_CHECK_PARAMS is not enabled, no test
+ *          can be made.
+ *
+ *          This macro is intended for negative parameter validation tests,
+ *          where the failing function can only return an error by calling
+ *          MBEDTLS_PARAM_FAIL to indicate the error.
+ *
+ * \param   TEST                The test expression to be tested.
+ */
+#define TEST_INVALID_PARAM( TEST ) \
+    do {                                                \
+        if ( setjmp( param_fail_jmp ) == 0 )            \
+        {                                               \
+            TEST;                                       \
+            test_fail( #TEST, __LINE__, __FILE__ );     \
+            goto exit;                                  \
+       }                                                \
+       memset( param_fail_jmp, 0, sizeof(jmp_buf) );    \
+    } while( 0 )
+
+#else
+
+#define TEST_ASSERT( TEST )                             \
+    do {                                                \
+        if( ! (TEST) )                                  \
+        {                                               \
+            test_fail( #TEST, __LINE__, __FILE__ );     \
+            goto exit;                                  \
+        }                                               \
+    } while( 0 )
+
+#define TEST_FN( TEST )                                 \
+    do {                                                \
+            TEST;                                       \
+    } while( 0 )
+
+#define TEST_INVALID_PARAM_RET( PARAM_ERR_VALUE, TEST ) \
```

Comment:
This might collide with [TEST_ASSERT_RET](https://github.com/ARMmbed/mbedtls/pull/2124/files#diff-5c77e3830486ba0964d963b0b85814ddR73).
Let's make this more generic, and rename to `TEST_ASSERT_RET`.